pandas/tests/series/test_missing.py

```diff
@@ -1204,6 +1204,41 @@ def test_interp_limit_to_ends(self):
                                limit_direction='both')
         assert_series_equal(result, expected)
 
+    def test_interp_maxgap(self):
+        s = Series([
+            np.nan,
+            1., np.nan,
+            2., np.nan, np.nan,
+            5., np.nan, np.nan, np.nan,
+            -1., np.nan, np.nan
+        ])
+
+        excpected = Series([
+            1.,
+            1., 1.5,
+            2., 3., 4.,
+            5., np.nan, np.nan, np.nan,
+            -1., -1, -1
+        ])
+
+        result = s.interpolate(method='linear', maxgap=2)
+        assert_series_equal(result, excpected)
+
+        excpected = Series([
+            np.nan,
+            1., 1.5,
+            2., 3., 4.,
+            5., np.nan, np.nan, np.nan,
+            -1., np.nan, np.nan
+        ])
+
+        result = s.interpolate(method='linear', maxgap=2, limit_area='inside')
+        assert_series_equal(result, excpected)
+
+        with pytest.raises(ValueError,
```

Comment:
can you put the fail cases in another test function; i think you need a bit more checking (as you have multiple fail conditions)